security/azure/sql_publicly_accessible_managed_instance/check_for_publicly_accessible_azure_sql_managed_instance.pt

```diff
@@ -98,11 +109,80 @@ datasource "ds_list_sql_managed_instances" do
       field "instance_state", jmes_path(col_item,"properties.state")
       field "isPublicDataEndpointEnabled", jmes_path(col_item,"properties.publicDataEndpointEnabled")
       field "tags", jmes_path(col_item,"tags")
-      field "tagKeys", jmes_path(col_item,"tags.keys(@)")
+	  field "subscriptionId",val(iter_item,"subscriptionId")
+      field "subscriptionName",val(iter_item,"displayName")
     end
   end
 end
 
+datasource "ds_sql_managed_instances" do
+  run_script $js_sql_managed_instances, $ds_list_sql_managed_instances, $param_exclusion_tag_key
+end
+
+##################
+# Script         #
+##################
+
+script "js_sql_managed_instances", type: "javascript" do
+  parameters "ds_list_sql_managed_instances", "param_exclusion_tag_key"
+  result "res"
+  code <<-EOS
+    res = []
+    for (var j = 0; j < ds_list_sql_managed_instances.length; j++) {
+      var instance = ds_list_sql_managed_instances[j];
+      var tags=instance['tags'];
+      var is_tag_matched=false;
+      var param_exclusion_tag_key_lowercase = param_exclusion_tag_key.toLowerCase();
+        if(typeof(tags) != "undefined" && tags!=null){
+          for(var k=0;k<tags.length;k++){
+            var tag=tags[k];
+              if(param_exclusion_tag_key_lowercase.indexOf((tag['tagKey']).toLowerCase()) !== -1){
+               is_tag_matched = true;
+              }
+          }
+        }         
+        if(!is_tag_matched){
+          res.push({
+            instance_state:instance['instance_state'],
+            name:instance['name'],
+            location:instance['location'],
+            license_type:instance['license_type'],
+            id:instance['id'],
+            v_cores:instance['v_cores'],
+            storage_size_in_gb:instance['storage_size_in_gb'],
+            isPublicDataEndpointEnabled:instance['isPublicDataEndpointEnabled'],
+            tags:instance['tags'],
+            subscriptionId:instance['subscriptionId'],
+            subscriptionName:instance['subscriptionName']
+          })
+        }
+    }
+  EOS
+end
+
+###############################################################################
+# Policy
+###############################################################################
+
+policy "pol_sql_managed_instances" do
+  validate_each $ds_sql_managed_instances do
+    summary_template "{{ rs_project_name }} (Account ID: {{ rs_project_id }}): {{ len data }} Azure SQL managed instances with public endpoint enabled"
+    detail_template <<-EOS
+# Azure SQL managed Instance Details
```

Comment:
Add sorting by subscription name and region.

Reply:
Sorting logic has been added